api/phone.go

```diff
@@ -29,39 +34,51 @@ func (a *API) formatPhoneNumber(phone string) string {
 	return strings.ReplaceAll(strings.Trim(phone, "+"), " ", "")
 }
 
-func (a *API) sendPhoneConfirmation(ctx context.Context, tx *storage.Connection, user *models.User, phone string) error {
+// sendPhoneConfirmation sends an otp to the user's phone number
+func (a *API) sendPhoneConfirmation(ctx context.Context, tx *storage.Connection, user *models.User, phone, otpType string, smsProvider sms_provider.SmsProvider) error {
 	config := a.getConfig(ctx)
 
-	if user.ConfirmationSentAt != nil && !user.ConfirmationSentAt.Add(config.Sms.MaxFrequency).Before(time.Now()) {
+	var token *string
+	var sentAt *time.Time
+	var tokenDbField, sentAtDbField string
+
+	if otpType == phoneConfirmationOtp {
+		token = &user.ConfirmationToken
+		sentAt = user.ConfirmationSentAt
+		tokenDbField, sentAtDbField = "confirmation_token", "confirmation_sent_at"
+	} else if otpType == phoneChangeOtp {
+		token = &user.PhoneChangeToken
+		sentAt = user.PhoneChangeSentAt
+		tokenDbField, sentAtDbField = "phone_change_token", "phone_change_sent_at"
+	} else {
+		return internalServerError("invalid otp type")
+	}
+
+	if sentAt != nil && !sentAt.Add(config.Sms.MaxFrequency).Before(time.Now()) {
```

Comment:
/nit 

personal preference but something of the style `(time.Now()).After(sentAt.Add(config.Sms.MaxFrequency))`  might be slightly more readable.  Not sure that this compiles though.

Not expecting any change, just leaving it here as another possibility to consider